src/components/checkbox/index.test.js

```diff
@@ -24,7 +24,10 @@ describe('Checkbox', () => {
     checked,
   }
 
-  const wrapper = shallow(<Checkbox {...props} />, { context: { theme } })
+  let wrapper
```

Comment:
This is closer to the pattern we'd like! Basically, what should be done here is to only mount (full or shallow) when necessary, as it is an expensive operation and want to do so only when we need to. Remember that `describe` blocks can be nested, so you can keep the existing ones if you'd like, but nest them in an over-arching `describe`.

So, for all the tests that don't contain manipulation of mocks (ie checking if things have been called), they could be grouped together in one `describe` block with a single `beforeAll`, meaning that the mounting of the component is only done once for all these tests.

Then we could have another `describe` block that deals with the manipulation and, assigning the `wrapper` in a `beforeEach`. With this approach, we will end up mounting fewer times and increase the readability and efficiency of the code 🙂

If any of this is unclear, ping me and we can work through it!